### PR TITLE
Humanize for DateTime? defaulting to "never"

### PR DESCRIPTION
**Pull request by rdoubleui**
In continuation to #479 I added the `Humanize` overload for `DateTime?`.

### CODE REVIEW

File: src/Humanizer.Tests/project.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-        "xunit.runner.visualstudio": "2.1.0"
-    },
+  "dependencies": {
+    "ApiApprover": "3.0.1",
+    "ApprovalTests": "3.0.10",
+    "ApprovalUtilities": "3.0.10",
+    "Mono.Cecil": "0.9.6.1",
+    "xunit": "2.1.0",
+    "xunit.runner.visualstudio": "2.1.0"
+  },
```
Comment (mexx) [inline src/Humanizer.Tests/project.json]: @rdoubleui can you revert the changes to this file, they are irrelevant to the feature, right?

File: src/Humanizer/DateTimeHumanizeStrategy/IDateTimeHumanizeStrategy.cs [hunk 2]
```diff
@@ -12,5 +12,10 @@ public interface IDateTimeHumanizeStrategy
         /// Calculates the distance of time in words between two provided dates used for DateTime.Humanize
         /// </summary>
         string Humanize(DateTime input, DateTime comparisonBase, CultureInfo culture);
+
+        /// <summary>
+        /// Calculates the distance of time in words between two provided dates used for DateTime?.Humanize
+        /// </summary>
+        string Humanize(DateTime? input, DateTime comparisonBase, CultureInfo culture);
```
Comment (mexx) [inline src/Humanizer/DateTimeHumanizeStrategy/IDateTimeHumanizeStrategy.cs]: I don't know whether we really need to have this method on `Strategy`.
As far as I can currently see, it would always have the same implementation:
```
if (input.HasValue)
  return Humanize(input.Value, comparisonBase, culture);
else
  return Configurator.GetFormatter(culture).DateHumanize_Never();
```

I would implement this directly in the `DateHumanizeExtensions` class.
By this the impact on the API by this feature would be minimized, and if someday there will be a need to have different strategies for `null` values, we can implement it the way currently shown here.

File: src/Humanizer/DateTimeHumanizeStrategy/IDateTimeOffsetHumanizeStrategy.cs [hunk 3]
```diff
@@ -12,5 +12,10 @@ public interface IDateTimeOffsetHumanizeStrategy
         /// Calculates the distance of time in words between two provided dates used for DateTimeOffset.Humanize
         /// </summary>
         string Humanize(DateTimeOffset input, DateTimeOffset comparisonBase, CultureInfo culture);
+
+        /// <summary>
+        /// Calculates the distance of time in words between two provided dates used for DateTimeOffset?.Humanize
+        /// </summary>
+        string Humanize(DateTimeOffset? input, DateTimeOffset comparisonBase, CultureInfo culture);
```
Comment (mexx) [inline src/Humanizer/DateTimeHumanizeStrategy/IDateTimeOffsetHumanizeStrategy.cs]: @rdoubleui Same as in `IDateTimeHumanizeStrategy`

File: src/Humanizer.Tests/project.json (hunk 1 above)
Comment (rdoubleui) [inline src/Humanizer.Tests/project.json]: Yes, of course.

File: src/Humanizer.Tests/ApiApprover/PublicApiApprovalTest.approve_public_api.approved.txt [hunk 4]
```diff
@@ -1028,12 +1030,12 @@ namespace Humanizer.DateTimeHumanizeStrategy
     }
     public class PrecisionDateTimeHumanizeStrategy : Humanizer.DateTimeHumanizeStrategy.IDateTimeHumanizeStrategy
     {
-        public PrecisionDateTimeHumanizeStrategy(double precision = 0.75) { }
+        public PrecisionDateTimeHumanizeStrategy(double precision = 0,75) { }
```
Comment (mexx) [inline src/Humanizer.Tests/ApiApprover/PublicApiApprovalTest.approve_public_api.approved.txt]: There is still no upstream version for this problem? :-( This line should be reverted, and 1038 also.

File: src/Humanizer/Localisation/Formatters/IFormatter.cs [hunk 5]
```diff
@@ -34,6 +40,6 @@ public interface IFormatter
         /// <param name="timeUnit"></param>
         /// <param name="unit"></param>
         /// <returns></returns>
-        string TimeSpanHumanize(TimeUnit timeUnit, int unit);
+        string TimeSpanHumanize(TimeUnit timeUnit, int unit);
```
Comment (mexx) [inline src/Humanizer/Localisation/Formatters/IFormatter.cs]: Please remove this extra whitespace at the end of the line.

File: src/Humanizer.Tests/ApiApprover/PublicApiApprovalTest.approve_public_api.approved.txt (hunk 4 above)
Comment (mexx) [inline src/Humanizer.Tests/ApiApprover/PublicApiApprovalTest.approve_public_api.approved.txt]: That was fast :)

Comment (rdoubleui) [inline src/Humanizer.Tests/ApiApprover/PublicApiApprovalTest.approve_public_api.approved.txt]: Done. The PR is still [pending] (https://github.com/JakeGinnivan/ApiApprover/pull/13).

Comment (rdoubleui) [inline src/Humanizer.Tests/ApiApprover/PublicApiApprovalTest.approve_public_api.approved.txt]: Yes :) Just saw the trouble within diff in that moment.

**Comment by mexx** [comment]
LGTM
@onovotny what do you say?

**Comment by onovotny** [comment]
:+1:

**Comment by hazzik** [comment]
Need to localize for supported languages

**Comment by rdoubleui** [comment]
@hazzik Yes, should this be another issue? How's this to be addressed?

**Comment by hazzik** [comment]
@rdoubleui another issue. How we will handle it? - not sure yet.
